File: src/main/java/jp/co/cybird/barcodekanojoForGAM/core/http/NameValueOrFilePair.java
package jp.co.cybird.barcodekanojoForGAM.core.http;

import java.io.File;
import org.apache.http.message.BasicNameValuePair;

public class NameValueOrFilePair extends BasicNameValuePair {
    private File file;

    public NameValueOrFilePair(String name, String value) {
        super(name, value);
    }

    public NameValueOrFilePair(String name, File file2) {
        super(name, null);
        this.file = file2;
    }

    public File getFile() {
        return this.file;
    }
}

File: src/main/java/com/goujer/barcodekanojo/core/http/NameValueOrFilePair.kt
package com.goujer.barcodekanojo.core.http

import java.io.File
import java.io.FileInputStream
import java.io.InputStream
import java.lang.StringBuilder
import java.net.URLEncoder

class NameValueOrFilePair: NameValuePair {
	internal var file: File?
		private set

	constructor(name: String, file: File?): super(name, null) {
		this.file = file
	}

	constructor(name: String, value: String?): super(name, value) {
		file = null
	}
}